modules/python-modules/README.md

```diff
@@ -0,0 +1,648 @@
+    earlier message may be acknowledged later than a message originally
+    encountered later in the source stream.
+
+  * consecutive tracking (ConsecutiveAckTracker): messages are assumed to
+    form a stream and the bookmark to be a position in that stream.
+    Unordered deliveries are properly handled by only acknowledging messages
+    that were delivered in order.  If an unordered delivery happens, the
+    tracker waits for the sequence to fill up, e.g.  waits all preceeding
+    messages to be delivered as well.
+
+  * batched tracking (BatchedAckTracker): messages are assumed to be
+    independent, not forming a sequence of events.  Each message is
+    individually tracked, the source driver has the means to get delivery
+    notifications of each and every message independently.  The
+    acknowledgements are accumulated until a timeout happens, at which point
+    they get reported as a single batch.
+
+You can initialize your ack_tracker in the `init` method, like this:
+
+```python
+
+class MyFetcher(LogFetcher):
+
+    ...
+
+    def init(self, options):
+        # pass self.message_acked method as ACK callback
+        self.ack_tracker = ConsecutiveAckTracker(ack_callback=self.message_acked)
+        return True
+
+    def message_acked(self, acked_message_bookmark):
+        pass
+
+    def fetch(self):
+        ...
+        msg.set_bookmark("whatever-bookmark-value-that-denotes-position")
+    ...
+
+```
+
+In the case above, we were using ConsecutiveAckTracker, e.g.  we would only
+get acknowledgements in the order messages were generated.  The argument of
+the `message_acked` callback would be the "bookmark" value that we set using
+set_bookmark().
+
+Using InstantAckTracker is very similar, just replace
+`ConsecutiveAckTracker` with `InstantAckTracker`. In this case you'd get a
+callback as soon as a message is delivered without preserving the original
+ordering.
+
+```python
+
+class MyFetcher(LogFetcher):
+
+    ...
+
+    def init(self, options):
+        self.ack_tracker = InstantAckTracker(ack_callback=self.message_acked)
+        return True
+
+    def message_acked(self, acked_message_bookmark):
+        pass
+```
+
+While ConsecutiveAckTracker() seems to provide a much more useful service,
+but InstantAckTracker() performs better, as it does not have to track
+acknowledgements of individual messages.
+
+The most complex scenario is implemented by BatchedAckTracker, this allows
+you to track the acks for individual messages, as they happen, not enforcing
+any kind of ordering.
+
+```python
+
+class MyFetcher(LogFetcher):
+
+    ...
+
+    def init(self, options):
+        self.ack_tracker = BatchedAckTracker(timeout=500, batch_size=100,
+                                             batched_ack_callback=self.messages_acked)
+        return True
+
+    def messages_acked(self, acked_message_bookmarks):
+        pass
+```
+
+BatchedAckTracker would call your callback every now and then, as specified
```

Comment:
[optional]

BatchedAckTracker is souce code, so it could be backticked.
```suggestion
`BatchedAckTracker` would call your callback every now and then, as specified
```